umm-spec-lib/src/cmr/umm_spec/test/umm_generators.clj

```diff
@@ -38,11 +39,19 @@
   with keys properties, required, and additionalProperties. This is used to handle a normal object
   with properties or an object which uses oneOf to specify between lists of properties."
   [schema type-name schema-type]
-  (rejected-unexpected-fields #{:properties :required :additionalProperties :not} schema-type)
+  (rejected-unexpected-fields #{:properties :required :additionalProperties :not :allOf} schema-type)
   (let [constructor-fn (if type-name
                          (record-gen/schema-type-constructor schema type-name)
                          identity)
         properties (:properties schema-type)
+        ;; The following line is a hack!!!
+        ;; Since UMM-G InstrumentType has ComposedOf field that is also an InstrumentType
+        ;; and there could be multiple levels of ComposedOf within an InstrumentType,
+        ;; We randomly (1 out of 2) drop the ComposedOf field from properties
+        ;; to not get into infinite loops of generating InstrumentTypes
+        properties (if (= 0 (rand-int 2))
```

Comment:
I don't quite follow why we had to use random as opposed to always picking one. Are you sure there is not a legitimate case that this code would not handle?

Reply:
Oh this is a generator for test cases... Ok I'm not so worried anymore.

Reply:
yeah, It just generates the ComposedOf part of the InstrumentType 50% of the time for testing. This way we can have a variety of test data that could go down multiple levels on the ComposedOf. It is a good hack. :)